spi_nand_flash/include/spi_nand_flash.h

```diff
@@ -24,6 +24,7 @@ struct spi_nand_flash_config_t {
     spi_device_handle_t device_handle;       ///< SPI Device for this nand chip.
     uint8_t gc_factor;                       ///< The gc factor controls the number of blocks to spare block ratio.
     ///< Lower values will reduce the available space but increase performance
+    uint8_t flags;                          ///< set flag with SPI_DEVICE_HALFDUPLEX for half duplex communcation.
```

Comment:
I wonder if this can be replaced with an enum (SIO/DIO/DOUT/QIO/QOUT) defined in this component, so that there is less dependency on the way SPI master driver defines its flags.